Core/include/Acts/Seeding/HoughTransformUtils.ipp

```diff
@@ -207,33 +239,42 @@ Acts::HoughTransformUtils::PeakFinders::IslandsAroundMax<
   // and obtain the fraction of the max that is our cutoff for island formation
   YieldType min = std::max(m_cfg.threshold, m_cfg.fractionCutoff * max);
   // book a list for the candidates and the maxima
-  std::vector<std::pair<std::size_t, std::size_t>> candidates;
+  const std::unordered_set<std::size_t>& nonEmptyBins{plane.getNonEmptyBins()};
+  std::vector<std::size_t> candidates;
+  candidates.reserve(nonEmptyBins.size());
   std::vector<Maximum> maxima;
+  maxima.reserve(nonEmptyBins.size());
   // keep track of the yields in each non empty cell
-  std::map<std::pair<std::size_t, std::size_t>, YieldType> yieldMap;
+  std::unordered_map<std::size_t, YieldType> yieldMap;
 
   // now loop over all non empty bins
-  for (auto [x, y] : plane.getNonEmptyBins()) {
-    // we only consider cells above threshold from now on.
-    // each is a potential candidate to seed or join an island
-    // We also add each to our yield map
-    if (plane.nHits(x, y) > min) {
-      candidates.push_back({x, y});
-      yieldMap[{x, y}] = plane.nHits(x, y);
+  for (const std::size_t nbin : nonEmptyBins) {
+    // std::array<std::size_t,2> xy = plane.axisBins(nbin);
+    //  we only consider cells above threshold from now on.
+    //  each is a potential candidate to seed or join an island
+    //  We also add each to our yield map
+
+    if (plane.nHits(nbin) > min) {
+      candidates.push_back(nbin);
+      yieldMap[nbin] = plane.nHits(nbin);
     }
   }
   // sort the candidate cells descending in content
   std::sort(candidates.begin(), candidates.end(),
-            [&plane](const std::pair<std::size_t, std::size_t>& bin1,
-                     const std::pair<std::size_t, std::size_t>& bin2) {
-              return (plane.nHits(bin1.first, bin1.second) >
-                      plane.nHits(bin2.first, bin2.second));
+            [&yieldMap](const std::size_t bin1, const std::size_t bin2) {
+              YieldType h1 = yieldMap[bin1];
+              YieldType h2 = yieldMap[bin2];
+
+              if (h1 != h2) {
+                return h1 > h2;
+              }
+              return bin1 > bin2;
```

Comment:
this is probably the non-stability in the maximum determination, because we now sort by global index for equal yields (which we did not in the past) ?

Reply:
yes right